### PR TITLE
WIP: Refactor javascript doc

### PR DESCRIPTION
**Pull request by benkoshy**
* Work in progress - this is a test case of the types of changes you want?
* If any objects pls post. (I did shorten a few words)

### CODE REVIEW

**Comment by ddnexus** [comment]
Hey Ben, 
looking at the page in retype... that looks soo full of stuff. I should have splitted it in sub pages long ago, but the old layout was so difficult to change!

I noticed that the content section on the right misses the big headlines, which kind of makes sense since the content page should not be an addition of many pages, as it is now.

The page at https://github.com/ddnexus/pagy/tree/master/lib/javascripts that you started to integrate, could go as a subpage as well, and the files in that dir can be linked, so that's convenient. 

What do you think?

**Comment by benkoshy** [comment]
> Hey Ben,
> looking at the page in retype... that looks soo full of stuff. I should have splitted it in sub pages long ago, but the old layout was so difficult to change!

Agreed. The length might confuse / scare readers.

> I noticed that the content section on the right misses the big headlines, which kind of makes sense since the content page should not be an addition of many pages, as it is now.
> The page at https://github.com/ddnexus/pagy/tree/master/lib/javascripts that you started to integrate, could go as a subpage as well, and the files in that dir can be linked, so that's convenient.
> What do you think?

My view is to avoid mixing github readmes and the documentation. i would merge them and get rid of the github javascript readme. Split into two separate pages: page A and page B?

**Page A: answer the following questions**
**(1)** What do the Javascript helpers give me?
    - List the helpers.
    - Show the helpers: visual images.
**(2)** How do I install it?
<hr>

**Page B: answers this question:**

**(3)** API page. (How does it work)?

In other words, separate:

(1) the what / how from the
(2) specific API details. 

### Questions:
* Your thoughts? Is this a ridiculous idea?
* Is it worth testing this out for a single page i.e. the `javascript.md`. or no?
* Also if you noticed: I elided some of words. snuck it in there. agree / disagree?

**Comment by ddnexus** [comment]
What about:

- Javascript (content of the Overview section, general concepts, links to details, ... 
  - Navs
  - Combo Navs
  - Files (Content of js files readme, maybe a better title than "Files")
  - AJAX

**Comment by benkoshy** [comment]
> What about:
> 
>     * Javascript (content of the Overview section, general concepts, links to details, ...
>       
>       * Navs
>       * Combo Navs
>       * Files (Content of js files readme, maybe a better title than "Files")
>       * AJAX

Yes much merit in this structure. I have in my mind a variation of this.

Here's my proposal. lemme have at  the `javascript` docs carte blanche . If the ideas don't work out: scrap them. Or adopt / change what works. any objections to this?

**Comment by ddnexus** [comment]
It's fine Ben. Seeing what you propose will probably be a good option. Thank you
